azurerm/data_source_kubernetes_cluster.go

```diff
@@ -564,6 +587,22 @@ func flattenKubernetesClusterDataSourceAgentPoolProfiles(input *[]containerservi
 			agentPoolProfile["count"] = int(*profile.Count)
 		}
 
+		if profile.MinCount != nil {
+			agentPoolProfile["min_count"] = int(*profile.MinCount)
+		}
+
+		if profile.MaxCount != nil {
+			agentPoolProfile["max_count"] = int(*profile.MaxCount)
+		}
+
+		if profile.EnableAutoScaling != nil {
+			agentPoolProfile["enable_auto_scaling"] = *profile.EnableAutoScaling
+		}
+
+		if profile.AvailabilityZones != nil {
+			agentPoolProfile["availability_zones"] = utils.FlattenStringSlice(profile.AvailabilityZones)
+		}
```

Comment:
@tombuildsstuff:
I was trying to avoid creating emepy entries in the map (agentPoolProfile). If profile.AvailabilityZones is nil, then we'll get an empty entry on the map. [Snippet for better explanation](https://play.golang.org/p/6Trfl-loUEf) 
Let me know what you think and I'll do it as instructed.

Reply:
As this is on the TF side what is the issue if we set this to an empty array?

Reply:
In the data source none. The rest of the code in that provider is following that pattern. Not sure if nicer in the state files (if it actually goes there). So I'll remove it, if you want it back just let me know.